modules/charon-core/src/main/java/org/wso2/charon3/core/schema/SCIMSchemaDefinitions.java

```diff
@@ -1643,6 +1643,43 @@ public static class SCIMResourceTypeSchemaDefinition {
                     SCIMRoleSchemaDefinition.ASSOCIATED_APPLICATIONS,
                     SCIMRoleSchemaDefinition.ROLE_V2_PROPERTIES);
 
+    /**
+     * RoleV3 Resource Schema.
+     */
+    public static final SCIMResourceTypeSchema SCIM_ROLE_V3_SCHEMA =
+            SCIMResourceTypeSchema.createSCIMResourceSchema(
+                    new ArrayList<>(Collections.singletonList(SCIMConstants.ROLE_SCHEMA_URI)),
+                    ID,
+                    SCIMRoleSchemaDefinition.DISPLAY_NAME,
+                    SCIMRoleSchemaDefinition.AUDIENCE,
+                    SCIMRoleSchemaDefinition.USERS,
+                    SCIMRoleSchemaDefinition.GROUPS,
+                    SCIMRoleSchemaDefinition.ROLE_V2_PERMISSIONS,
+                    SCIMRoleSchemaDefinition.ASSOCIATED_APPLICATIONS,
+                    SCIMRoleSchemaDefinition.ROLE_V2_PROPERTIES);
+
+    /**
+     * RoleV3 Resource Schema for User.
+     */
+    public static final SCIMResourceTypeSchema SCIM_ROLE_V3_USER_SCHEMA =
+            SCIMResourceTypeSchema.createSCIMResourceSchema(
+                    new ArrayList<>(Collections.singletonList(SCIMConstants.ROLE_SCHEMA_URI)),
+                    ID,
+                    SCIMRoleSchemaDefinition.USERS,
+                    SCIMRoleSchemaDefinition.ASSOCIATED_APPLICATIONS,
+                    SCIMRoleSchemaDefinition.ROLE_V2_PROPERTIES);
+
+    /**
+     * RoleV3 Resource Schema for Groups.
+     */
+    public static final SCIMResourceTypeSchema SCIM_ROLE_V3_GROUPS_SCHEMA =
+            SCIMResourceTypeSchema.createSCIMResourceSchema(
+                    new ArrayList<>(Collections.singletonList(SCIMConstants.ROLE_SCHEMA_URI)),
+                    ID,
+                    SCIMRoleSchemaDefinition.GROUPS,
+                    SCIMRoleSchemaDefinition.ASSOCIATED_APPLICATIONS,
+                    SCIMRoleSchemaDefinition.ROLE_V2_PROPERTIES);
```

Comment:
What is the objective of defining this schema.
In this class, we need to maintain the scim resources' attribute schema.

role's user and role's group are just attributes of the scim role resource but not entirely a scim resource.
So it's not appropriate to define schema for them.
If you need to do validations on role's user assignment and role's group assignment level, that should be a validation in API level

Reply:
Fixed with : https://github.com/wso2/charon/pull/435/commits/adb417717e375a2b27daa8200ee3b635f49e8332